be/src/exec/pipeline/nljoin/spillable_nljoin_probe_operator.h

```diff
@@ -0,0 +1,192 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <memory>
+
+#include "column/vectorized_fwd.h"
+#include "exec/pipeline/nljoin/nljoin_context.h"
+#include "exec/pipeline/nljoin/nljoin_probe_operator.h"
+#include "exec/pipeline/operator_with_dependency.h"
+#include "exec/spill/executor.h"
+#include "exec/spill/spiller_factory.h"
+#include "runtime/runtime_state.h"
+#include "util/runtime_profile.h"
+
+namespace starrocks::pipeline {
+
+class NLJoinProber {
```

Comment:
can NLJoinProber be shared by NLJoinProbeOperator? seems some logics are similar

Reply:
Our goal is to be able to replace NLJoinProbeOperator, but not all features are currently supported